File: Rubik.HTML/Server/Schema/CubeInfo.cs
using System;
using System.Web;
using System.ComponentModel;
using System.Collections.Generic;

namespace Rubik.Server.Schema
{
    public class CubeInfo : InfoObject
    {
        public List<DimensionInfo> Dimensions
        {
            get; set;
        }

        public List<MeasureInfo> Measures
        {
            get; set;
        }
    }
}

File: Rubik.HTML/Lib/DataHub/Schema/CubeInfo.ts
module Rubik.DataHub {
    export class CubeInfo extends InfoObject {
        get ObjectType(): ObjectTypeEnum { return ObjectTypeEnum.Cube; }

        get Dimensions(): Promise<DimensionInfo[]> {
            return this.factory.GetInfoObjectCollection(Rubik.DataHub.ObjectTypeEnum.Dimension, this) as Promise<DimensionInfo[]>;
        }
        get Hierarchies(): Promise<HierarchyInfo[]> {
            return this.factory.GetInfoObjectCollection(Rubik.DataHub.ObjectTypeEnum.Hierarchy, this) as Promise<HierarchyInfo[]>;
        }
        get Measures(): Promise<MeasureInfo[]> {
            return this.factory.GetInfoObjectCollection(Rubik.DataHub.ObjectTypeEnum.Measure, this) as Promise<MeasureInfo[]>;                    
        }
    }
}